reactive/jenkins.py

```diff
@@ -230,3 +248,34 @@ def update_nrpe_config(nagios):
 def stop():
     service_stop("jenkins")
     status_set("maintenance", "Jenkins stopped")
+
+
+@when("reset.jenkins.storage")
+def reset_jenkins_storage():
+    status_set("maintenance", "Configuring Jenkins Storage")
+
+    jenkins_installed = get_state("apt.installed.jenkins")
+    if jenkins_installed:
+        service_stop('jenkins')
+
+    storage_dir = unitdata.kv().get("storage.dir")
+    if storage_dir is None:
+        log("Setting Jenkins to use local storage")
+        paths.Storage().unlink_home()
+    else:
+        log("Setting Jenkins to use storage at {}".format(storage_dir))
+        paths.Storage().link_home(storage_dir)
+
+    if jenkins_installed:
+        status_set("maintenance", "Restarting Jenkins")
+        service_start('jenkins')
+        Service().check_ready()
+
+    remove_state("reset.jenkins.storage")
+
+    if get_state('jenkins.bootstrapped'):
+        # JENKINS_HOME just changed trigger bootstrap again
+        remove_state("jenkins.bootstrapped")
+        bootstrap_jenkins()
+    else:
+        status_set('active', 'Ready')
```

Comment:
No need to trigger a bootstrap if it hasn't been done yet, the next Juju hook should trigger it, that is the else case. The status_set was simply to leave this hook in the default state again rather than have it still say, 'Configuring Jenkins Storage'

The if state is because bootstrap needs to be re-triggered to correctly setup Jenkins if had previously been run. This will be encountered if you add storage to an already running application.